pkg/models/base_struct.go

```diff
@@ -0,0 +1,51 @@
+package models
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// IBaseStruct is an interface that all models must implement
+type IBaseStruct interface {
+	GetBaseStruct() *baseStruct
+	GetID() uuid.UUID
+	GetCreatedBy() string
+	GetModifiedBy() *string
+}
+
+// baseStruct represents the shared data in common betwen all models
+type baseStruct struct {
+	ID          uuid.UUID  `json:"id" db:"id"`
+	CreatedBy   string     `json:"createdBy" db:"created_by"`
+	CreatedDts  time.Time  `json:"createdDts" db:"created_dts"`
+	ModifiedBy  *string    `json:"modifiedBy" db:"modified_by"`
+	ModifiedDts *time.Time `json:"modifiedDts" db:"modified_dts"`
+}
+
+// NewBaseStruct returns a base struct object
+func NewBaseStruct(createdBy string) baseStruct {
+	return baseStruct{
```

Comment:
Should this maybe initialize `CreatedDts` as well?

Reply:
Good call out, created_dts is handled in the database insert statement, so even if we initialized it here, it wouldn't get written to the db.

Reply:
We could change the the approach here to initialize the time in the GO code if preferred too. It looks like most other places in the app code the time stamp is initialized from. FWIW, I compared the result with the time stamp from a system intake, vs initializing in the DB as done here, and the result is equivalent.